src/main/java/net/mcreator/ui/blockly/BlocklyPanel.java

```diff
@@ -173,6 +188,184 @@ public BlocklyPanel(MCreator mcreator, @Nonnull BlocklyEditorType type) {
 					runAfterLoaded.forEach(ThreadUtil::runOnFxThread);
 				}
 			});
+
+
+			// Quick search menu
+			browser.setOnMouseClicked(event -> {
+
+				if (event.getButton() == MouseButton.MIDDLE) {
+
+					class quickSearchMenuPopup extends JFrame {

```

Comment:
Externalize the class as it's a pretty big one (and you should rename it for `QuickSearchMenuPopup`, so it's the same format as everything else.